pmd-scala/src/test/java/net/sourceforge/pmd/scala/rule/ScalaRuleTest.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.scala.rule;
+
+import java.io.StringReader;
+import java.util.Arrays;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.LanguageRegistry;
+import net.sourceforge.pmd.lang.LanguageVersionHandler;
+import net.sourceforge.pmd.lang.Parser;
+import net.sourceforge.pmd.lang.scala.ScalaLanguageModule;
+import net.sourceforge.pmd.lang.scala.ast.ScalaNode;
+import net.sourceforge.pmd.lang.scala.rule.ScalaRule;
+
+public class ScalaRuleTest {
+    private static final String SCALA_TEST = "/parserFiles/helloworld.scala";
+
+    @Test
+    public void testRuleVisits() throws Exception {
+        LanguageVersionHandler scalaVersionHandler = LanguageRegistry.getLanguage(ScalaLanguageModule.NAME)
+                .getDefaultVersion().getLanguageVersionHandler();
+        Parser parser = scalaVersionHandler.getParser(scalaVersionHandler.getDefaultParserOptions());
+        ScalaNode root = (ScalaNode) parser.parse(null,
+                new StringReader(IOUtils.toString(getClass().getResourceAsStream(SCALA_TEST), "UTF-8")));
+
+        ScalaTestRule rule = new ScalaTestRule();
+        rule.apply(Arrays.asList(root), null);
+        Assert.assertEquals(12, rule.visited);
+    }
+
+    class ScalaTestRule extends ScalaRule {
+        int visited = 0;
+
+        @Override
+        public Object visit(ScalaNode node, Object data) {
```

Comment:
I'm not sure, whether this visitor is really helpful: If want to implement the XPath rule `//ASTTermApplyNode/ASTTermNameNode[@Image = 'println']` in Java, we probably need a couple of instanceof, like this:

```java
public Object visit(ScalaNode node, Object data) {
    Tree n = node.getNode();
    
    if (n instanceof TermApply) {
        int i = 0;
        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
            Node child = node.jjtGetChild(i);
            Tree c = child.getNode();
            if (c instanceof TermName && child.hasImageEqualTo("println")) {
                addViolation(data, child);
            }
        }
    }
    return data;
}
```

With a more expressive visitor AST inside PMD, it might look like this:
``` java
public Object visit(TermApply node, Object data) {
    Node child = node.getFirstChildOfType(TermName.class);
    if (child.hasImageEqualTo("println")) {
        addViolation(data, child);
    }
    return data;
}
```

Build such an AST however requires more work...